os_migrate/roles/export_users/tasks/main.yml

```diff
@@ -0,0 +1,37 @@
+- name: scan available users
+  os_user_info:
+    auth: "{{ os_migrate_src_auth }}"
+    auth_type: "{{ os_migrate_src_auth_type|default(omit) }}"
+    region_name: "{{ os_migrate_src_region_name|default(omit) }}"
+    validate_certs: "{{ os_migrate_src_validate_certs|default(omit) }}"
+    ca_cert: "{{ os_migrate_src_ca_cert|default(omit) }}"
+    client_cert: "{{ os_migrate_src_client_cert|default(omit) }}"
+    client_key: "{{ os_migrate_src_client_key|default(omit) }}"
+  register: src_users_info
+
+- name: create id-name pairs of users to export
+  set_fact:
+    export_users_ids_names: "{{ (
+      src_users_info.openstack_users
+        | json_query('[*].{name: name, id: id}')
+        | sort(attribute='id') ) }}"
+
+- name: filter names of users to export
+  set_fact:
+    export_users_ids_names: "{{ (
+      export_users_ids_names
+        | os_migrate.os_migrate.stringfilter(os_migrate_users_filter,
+                                             attribute='id') ) }}"
+
+- name: export users
+  os_migrate.os_migrate.export_user:
+    auth: "{{ os_migrate_src_auth }}"
+    auth_type: "{{ os_migrate_src_auth_type|default(omit) }}"
+    region_name: "{{ os_migrate_src_region_name|default(omit) }}"
+    path: "{{ os_migrate_data_dir }}/users.yml"
+    name: "{{ item['name'] }}"
```

Comment:
Should be item['id'] ?

Reply:
In case of users it might not matter, OpenStack allows duplicate naming on many resources but i doubt it would allow it on users (?), but still it would be nice to keep consistent with how we look up resources elsewhere.

Reply:
fixed, it was one of the changes I did for trying to see why I wasnt able to find the user